packages/lit-html/src/lib/lit-html.ts

```diff
@@ -597,7 +626,7 @@ export class NodePart {
     const directive = value._$litDirective$;
     if (this.__directive?.constructor !== directive) {
       this.__clear();
-      this.__directive = new directive(this);
+      this.__directive = new directive(this as NodePartInfo);
```

Comment:
Is this enough safety? You still get the real part.

Reply:
It's some help. We could generate an info object instead...